### PR TITLE
don't load audio repeatedly

### PR DESCRIPTION
**Pull request by milomg**

### CODE REVIEW

File: src/utils/midi.ts [hunk 1]
```diff
@@ -19,19 +18,23 @@ export interface MidiNote {
 function loadInstrument(instrument: string): Promise<boolean> {
```
Comment (1whatleytay) [inline src/utils/midi.ts:18]: This typing is wrong now, right?

File: src/utils/midi.ts [hunk 2]
```diff
@@ -41,6 +44,8 @@ export async function playNote(note: MidiNote) {
     if (!(await loadInstrument(note.name))) {
```
Comment (1whatleytay) [inline src/utils/midi.ts:44]: This return value probably shouldn't be checked? Is it awaiting `undefined` now or am I missing something?

File: src/utils/midi.ts [hunk 3]
```diff
@@ -41,6 +44,8 @@ export async function playNote(note: MidiNote) {
...
       return await wake()
     }
+  } else {
+    await loadedInstruments.get(note.name)
```
Comment (1whatleytay) [inline src/utils/midi.ts:48]: Not totally sure, but does anything bad happen here if this promise resolves to false? Should this abort?

**Review by 1whatleytay — COMMENTED** [review]
Just commenting some questions for now.

File: src/utils/midi.ts [hunk 4]
```diff
@@ -16,31 +15,37 @@ export interface MidiNote {
...
 export async function playNote(note: MidiNote) {
   const wake = async () => await tauri.invoke('wake_sync')
 
-  if (!loadedInstruments.has(note.name)) {
-    if (!(await loadInstrument(note.name))) {
-      return await wake()
-    }
+  if (
+    !loadedInstruments.has(note.name) ||
+    !(await loadedInstruments.get(note.name))
```
Comment (1whatleytay) [inline src/utils/midi.ts:45]: Are we using the value of undefined here? Still seems a bit strange.

**Review by 1whatleytay — APPROVED** [review]
Approving, still left a comment but it's minimal. Will merge tomorrow morning if I don't hear back/you don't want to change this.

File: src/utils/midi.ts (hunk 4 above)
Comment (modderme123) [reply src/utils/midi.ts:45]: The promise stored in the map resolves to true or false

**Pull request merged**